docs/guide/core-concepts.rst

```diff
@@ -558,435 +576,592 @@ possible, Constant ``NoData`` values are to be preferred. For
+    res3: String =
+    "    1
+         2
+         3
+    "
+
+    scala> res2.asciiDraw()
+    res4: String =
+    "    1     2     3
+         4     5     6
+         7     8     9
+    "
+
+Tile Hierarchy
+--------------
+
+For the sake of completeness, the following tile hierarchy is presented:
+
+.. figure:: ./images/tile-hierarchy.png
+   :alt:
+
+The ``Tile`` trait has operations you'd expect for traversing and
+transforming the contents of the tiles, like:
+
+-  ``map: (Int => Int) => Tile``
+-  ``foreach: (Int => Unit) => Unit``
+-  ``combine: Tile => ((Int, Int) => Int) => Tile``
+-  ``color: ColorMap => Tile``
+
+As discussed above, the ``Tile`` interface carries information about how big
+it is and what its underlying `Cell Type <#cell-types>`__ is:
+
+-  ``cols: Int``
+-  ``rows: Int``
+-  ``cellType: CellType``
 
 .. raw:: html
 
    <hr>
 
-Raster Data
-===========
 
-    “Yes raster is faster, but raster is vaster and vector just SEEMS
-    more corrector.” — `C. Dana
-    Tomlin <http://uregina.ca/piwowarj/NotableQuotables.html>`__
+Layouts and Tile Layers
+=======================
+
+The core vector and raster functionality thus far described stands on its own
+for small scale applications.  But, as mentioned, Geotrellis is intended to
+work with big data in a distributed context.  For this, we rely on Apache
+Spark's resilient distributed dataset (RDD).  RDDs of both raster and vector
+data are naturally supported by Geotrellis, but some new concepts are required
+to integrate this abstraction for distributed processing.
+
+For most applications, the data of interest must be keyed to a layout to give
+the content of an RDD—which is usually a collection of key-value pairs (i.e.,
+``RDD[(K, V)]``)—a consistent interpretation as a cohesive raster.  In such an
+RDD, the key type, ``K``, is one of ``TemporalKey``, ``SpatialKey``, or
+``SpaceTimeKey``.  The latter two key types obviously contain spatial data
+(declared in context bounds as ``[K: SpatialComponent]``, where values of such
+a type ``K`` can have their spatial component extracted using the
+``getComponent[SpatialKey]`` extension method), which is used to identify a
+region in space.
+
+The |LayoutDefinition|_ class is used to describe how
+``SpatialKey``\ s map to regions in space.  The ``LayoutDefinition`` is a
+``GridExtent`` subclass defined with an ``Extent`` and ``CellSize``.  The
+``Extent`` is subdivided into a grid of uniform, rectangular regions.  The
+size and number of the sub-regions is determined using the ``CellSize`` of the
+``LayoutDefinition``, and then the pixel dimensions of the constituent tiles.
+The sub-regions are then assigned a ``SpatialKey`` with the ``(0, 0)``
+position corresponding to the upper-left corner of the extent; the `x`
+coordinate increases toward the right, and the `y` coordinate increases moving
+down (into lower latitude values, say).
+
+.. |LayoutDefinition| replace:: ``geotrellis.spark.tiling.LayoutDefinition``
```

Comment:
It might be good to have `LayoutDefinition` be its own subsection since the `Layout Definition` link in the index brings the user to the `Layouts and Tile Layers` section.

Reply:
The `Layouts and Tile Layers` section is a fairly succinct description of LayoutDefinition and TileLayerMetadata, and it would, I think, unnecessarily bloat up the delivery of the concept to break it up into sections so that each entry of a glossary (which is not an index, and links are provided as more of a courtesy) can link to a unique section.  If there's an argument for restructuring that section into smaller sections to aid clarity, and it happens that the resulting structure makes for a 1-to-1 correspondence with glossary items, I'm happy to update the links, but I don't think having unique links is a good enough reason.

Reply:
I see your point. I think that section is pretty good as it is, so there probably is no reason to rewrite it at the moment. I'm not sure if it's possible, but I think that if you can link to the middle part of a section that would be best. Though, it's not really necessary.